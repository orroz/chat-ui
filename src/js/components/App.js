import React from "react";
import ConnectedMessagesList from "./MessagesList";
import ConnectedMessageBox from "./MessageBox";
import ConnectedUserBox from "./UserBox";
import ConnectedAvatar   from "./Avatar";

const App = () => (
	<div className="row mt-5">
		<div className="col-md-12 chatBox">
			<h2>Chat Room</h2>
			<ConnectedMessagesList />
		</div>
		<div className="col-md-12 controlBox">
			<div className="col-md-1">
				<ConnectedAvatar />
			</div>
			<div className="col-md-3">
				<h2>Whats Your Name</h2>
				<ConnectedUserBox />
			</div>
			<div className="col-md-6">
				<h2>Send A Message</h2>
				<ConnectedMessageBox />
			</div>
		</div>
	</div>
);
export default App;