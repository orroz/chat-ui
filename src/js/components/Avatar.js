// src/js/components/List.js
import React from "react";
import { connect } from "react-redux";
import avatarStore from "../modules/avatarStore";

const mapStateToProps = state => {
	return { avatarIndex: state.avatarIndex };
};
const ConnectedAvatarIcon = ({ avatarIndex }) => {
	let avatarUrl = avatarStore.getAvatarUrlByIndex(avatarIndex);
	return (<div className="avatarWrapper">
		<img className="avatarIcon" src={avatarUrl}></img>
	</div>)
};
const AvatarIcon = connect(mapStateToProps)(ConnectedAvatarIcon);
export default AvatarIcon;