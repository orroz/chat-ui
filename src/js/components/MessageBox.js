// src/js/components/Form.js
import React, { Component } from "react";
import { connect } from "react-redux";
import uuidv1 from "uuid";
import { addMessage } from "../actions/index";

const mapDispatchToProps = dispatch => {
	return {
		addMessage: message => dispatch(addMessage(message))
	};
};
export class MessageBox extends Component {
	constructor(props) {
		super(props);
		this.state = {
			text: ""
		};
		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}
	handleChange(event) {
		this.setState({ [event.target.id]: event.target.value });
	}
	handleSubmit(event) {
		event.preventDefault();
		if(!this.state.text){
			return;
		}
		const { text } = this.state;
		const id = uuidv1();
		this.props.addMessage({ text, id, sendOut: true });
		this.setState({ text: "" });
	}
	render() {
		const { text } = this.state;
		return (
			<form onSubmit={this.handleSubmit}>
				<div className="form-group">
					<input
						type="text"
						className="form-control"
						id="text"
						value={text}
						onChange={this.handleChange}
					/>
				</div>
				<button type="submit" className="sendButton btn btn-success btn-lg">
					Send
				</button>
			</form>
		);
	}
}
const ConnectedMessageBox = connect(null, mapDispatchToProps)(MessageBox);
export default ConnectedMessageBox;