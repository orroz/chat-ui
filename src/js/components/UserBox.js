// src/js/components/Form.js
import React, { Component } from "react";
import { connect } from "react-redux";
import { changeUsername } from "../actions/index";
const mapDispatchToProps = dispatch => {
	return {
		changeUsername: name => dispatch(changeUsername(name))
	};
};
const mapStateToProps = state => {
	return { name: state.name };
};
export class UserBox extends Component {
	constructor() {
		super();
		this.handleChange = this.handleChange.bind(this);
		this.state = {
			name : ""
		};
	}
	handleChange(event) {
		this.setState({ [event.target.id]: event.target.value });
		this.props.changeUsername(event.target.value);
	}
	componentWillMount(){
		console.log("componentWillMount:", this.props);
		this.setState({"name" : this.props.name});
	}
	render() {
		return (
				<div className="form-group">
					<input
						type="text"
						className="form-control"
						id="name"
						value={this.state.name}
						onChange={this.handleChange}
					/>
				</div>
		);
	}
}
const ConnectedUserBox = connect(mapStateToProps, mapDispatchToProps)(UserBox);
export default ConnectedUserBox;