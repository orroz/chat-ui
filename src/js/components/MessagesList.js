// src/js/components/List.js
import React, {Component} from "react";
import { connect } from "react-redux";
import {AvatarIcon} from "./Avatar";
const mapStateToProps = state => {
	return { messages: state.messages };
};

export class MessagesList extends Component {
	constructor(props) {
		super(props);
	};
	render() {
		let messages = this.props.messages || [];
		return ( <ul className="list-group list-group-flush">
				{messages.map(el => {
					let wrapperClass = el.isMe ? "isMe" : "isOther";
					return (<li className="list-group-item" key={el.id}>
						<div className={wrapperClass}><AvatarIcon avatarIndex={el.avatarIndex} />
							<span className="chatName">{el.name}</span> : <span className="chatContent">{el.text}</span></div>
					</li>);
				})}
			</ul>);
	}
}

const connectedMessagesList = connect(mapStateToProps)(MessagesList);
export default connectedMessagesList;