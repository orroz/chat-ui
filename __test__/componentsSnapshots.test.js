import React from 'react'
import renderer from 'react-test-renderer'
import {MessagesList} from '../src/js/components/MessagesList'
import {UserBox} from '../src/js/components/UserBox'
import {AvatarIcon} from '../src/js/components/Avatar'
import {MessageBox} from '../src/js/components/MessageBox'


const userName = "Or Rosenthal";
const mockMessage = { name: "mock message", avatarIndex : 1, title: "this is a mock message", id: 1, sendOut: false};

// Snapshot for MessageList React Component
describe('>>> MessagesList --- Snapshot comparison',()=>{
	it('+++capturing Snapshot of MessagesList with no messages', () => {
		const renderedValue =  renderer.create(<MessagesList />).toJSON();
		expect(renderedValue).toMatchSnapshot();
	});
	it('+++capturing Snapshot of MessagesList with one message', () => {
		const renderedValue =  renderer.create(<MessagesList messages={[mockMessage]} />).toJSON();
		expect(renderedValue).toMatchSnapshot();
	});
});

describe('>>> UserBox --- Snapshot comparison',()=>{
	it('+++capturing Snapshot of UserBox with no name', () => {
		const renderedValue =  renderer.create(<UserBox />).toJSON();
		expect(renderedValue).toMatchSnapshot();
	});
	it('+++capturing Snapshot of UserBox with user name', () => {
		const renderedValue =  renderer.create(<UserBox name={userName} />).toJSON();
		expect(renderedValue).toMatchSnapshot();
	});
});

describe('>>> AvatarIcon --- Snapshot comparison',()=>{
	it('+++capturing Snapshot of AvatarIcon with with icon index 2 - pikachu', () => {
		const renderedValue =  renderer.create(<AvatarIcon avatarIndex="2" />).toJSON();
		expect(renderedValue).toMatchSnapshot();
	});
});

describe('>>> MessageBox --- Snapshot comparison',()=>{
	it('+++capturing Snapshot of MessageBox', () => {
		const renderedValue =  renderer.create(<MessageBox />).toJSON();
		expect(renderedValue).toMatchSnapshot();
	});
});

