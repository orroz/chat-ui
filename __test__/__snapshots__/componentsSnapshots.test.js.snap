exports[`>>> AvatarIcon --- Snapshot comparison +++capturing Snapshot of AvatarIcon with with icon index 2 - pikachu 1`] = `
<div
  className="avatarWrapper">
  <img
    className="avatarIcon"
    src="https://spotim-demo-chat-server.herokuapp.com/avatars/003-pikachu.png" />
</div>
`;

exports[`>>> MessageBox --- Snapshot comparison +++capturing Snapshot of MessageBox 1`] = `
<form
  onSubmit={[Function]}>
  <div
    className="form-group">
    <input
      className="form-control"
      id="text"
      onChange={[Function]}
      type="text"
      value="" />
  </div>
  <button
    className="sendButton btn btn-success btn-lg"
    type="submit">
    Send
  </button>
</form>
`;

exports[`>>> MessagesList --- Snapshot comparison +++capturing Snapshot of MessagesList with no messages 1`] = `
<ul
  className="list-group list-group-flush" />
`;

exports[`>>> MessagesList --- Snapshot comparison +++capturing Snapshot of MessagesList with one message 1`] = `
<ul
  className="list-group list-group-flush">
  <li
    className="list-group-item">
    <div
      className="isOther">
      <div
        className="avatarWrapper">
        <img
          className="avatarIcon"
          src="https://spotim-demo-chat-server.herokuapp.com/avatars/002-psyduck.png" />
      </div>
      <span
        className="chatName">
        mock message
      </span>
       : 
      <span
        className="chatContent" />
    </div>
  </li>
</ul>
`;

exports[`>>> UserBox --- Snapshot comparison +++capturing Snapshot of UserBox with no name 1`] = `
<div
  className="form-group">
  <input
    className="form-control"
    id="name"
    onChange={[Function]}
    type="text"
    value={undefined} />
</div>
`;

exports[`>>> UserBox --- Snapshot comparison +++capturing Snapshot of UserBox with user name 1`] = `
<div
  className="form-group">
  <input
    className="form-control"
    id="name"
    onChange={[Function]}
    type="text"
    value="Or Rosenthal" />
</div>
`;
